## Setup

In [ ]:
import pandas as pd

In [ ]:
ufos_path = 'YOUR PATH HERE'
sightings = pd.read_csv(ufos_path)
sightings.head()

<ipython-input-2-828416dd11ab>:2: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  sightings = pd.read_csv(ufos_path)


datetime                  city state country     shape  \
0  10/10/1949 20:30            san marcos    tx      us  cylinder   
1  10/10/1949 21:00          lackland afb    tx     NaN     light   
2  10/10/1955 17:00  chester (uk/england)   NaN      gb    circle   
3  10/10/1956 21:00                  edna    tx      us    circle   
4  10/10/1960 20:00               kaneohe    hi      us     light   

  duration (seconds) duration (hours/min)  \
0               2700           45 minutes   
1               7200              1-2 hrs   
2                 20           20 seconds   
3                 20             1/2 hour   
4                900           15 minutes   

                                            comments date posted    latitude  \
0  This event took place in early fall around 194...   4/27/2004  29.8830556   
1  1949 Lackland AFB&#44 TX.  Lights racing acros...  12/16/2005    29.38421   
2  Green/Orange circular disc over Chester&#44 En...   1/21/2008        53.2   
3  My older brother and twin sister were leaving ...   1/17/2004  28.9783333   
4  AS a Marine 1st Lt. flying an FJ4B fighter/att...   1/22/2004  21.4180556   

    longitude  
0  -97.941111  
1  -98.581082  
2   -2.916667  
3  -96.645833  
4 -157.803611

## Use "or" logic to generate a filtered subset of UFO sightings that meet the following criteria:

* The sightings must have occurred in one of the following states: Washington (abbreviated "wa"), or Oregon ("or").

* OR, the sightings must have occurred between 33 and 38 degrees latitude.

In [ ]:
sightings['latitude'] = pd.to_numeric(sightings['latitude'], errors='coerce')

In [ ]:
wa_or = sightings['state'].isin(['wa','or'])
bt_33_38_lat = sightings['latitude'].between(33,38)

subset_1 = sightings[wa_or | bt_33_38_lat]
subset_1.head(10)

datetime                    city state country     shape  \
5   10/10/1961 19:00                 bristol    tn      us    sphere   
8   10/10/1966 20:00               pell city    al      us      disk   
10  10/10/1968 13:00               hawthorne    ca      us    circle   
11  10/10/1968 19:00                 brevard    nc      us  fireball   
13  10/10/1970 19:00              manchester    ky      us   unknown   
14  10/10/1971 21:00               lexington    nc      us      oval   
15  10/10/1972 19:00           harlan county    ky      us    circle   
23  10/10/1976 20:30               washougal    wa      us      oval   
44  10/10/1988 22:45  seattle (ballard area)    wa      us   unknown   
45  10/10/1989 00:00               calabasas    ca      us      disk   

   duration (seconds) duration (hours/min)  \
5                 300            5 minutes   
8                 180           3  minutes   
10                300               5 min.   
11                180            3 minutes   
13                180            3 minutes   
14                 30           30 seconds   
15               1200            20minutes   
23                 60             1 minute   
44                  2               2 sec.   
45                300         approx 5 min   

                                             comments date posted   latitude  \
5   My father is now 89 my brother 52 the girl wit...   4/27/2007  36.595000   
8   Strobe Lighted disk shape object observed clos...   3/19/2009  33.586111   
10  ROUND &#44 ORANGE &#44 WITH WHAT I WOULD SAY W...  10/31/2003  33.916389   
11  silent red /orange mass of energy floated by t...   6/12/2008  35.233333   
13  Slow moving&#44 silent craft accelerated at an...   2/14/2008  37.153611   
14  green oval shaped light over my local church&#...   2/14/2010  35.823889   
15  On october 10&#44 1972 myself&#44my 5yrs.daugh...   9/15/2005  36.843056   
23  Three extremely large lights hanging above nea...    2/7/2014  45.582778   
44  Two adult witnesses are stunned by the sound o...   4/27/2004  47.606389   
45         Unidentified object on Mulholland Highway.  12/14/2004  34.157778   

     longitude  
5   -82.188889  
8   -86.286111  
10 -118.351667  
11  -82.734444  
13  -83.761944  
14  -80.253611  
15  -83.321944  
23 -122.352222  
44 -122.330833  
45 -118.637500

## How many sightings met these criteria?

In [ ]:
len(subset_1)

26686

**26,686**

## Use "or" logic to generate a filtered subset of UFO sightings that meet the following criteria:

* The sightings must occurred either above 66.5 degrees latitude OR below -66.5 degrees latitude.

* OR, the sightings must have occurred in state of Colorado (abbreviated "co").

In [ ]:
# It's totally fine (and probably preferable) if you chose to assign each boolean series to a variable BTW; I just opted for the condensed approach in this case to keep you familiar with that syntax.

subset_2 = sightings[(sightings['latitude'] > 66.5) | (sightings['latitude'] < -66.5) | (sightings['state'] == 'co')]
subset_2.head(10)

datetime              city state country     shape  \
31   10/10/1979 22:00          security    co      us   unknown   
43   10/10/1988 22:00           boulder    co      us     light   
167  10/10/2007 04:00            denver    co      us  changing   
180  10/10/2007 22:00           fleming    co      us    circle   
183  10/10/2007 23:05        northglenn    co      us  triangle   
198  10/10/2008 20:40            pueblo    co      us     light   
203  10/10/2009 12:00     crested butte    co      us       egg   
252  10/10/2012 20:00  colorado springs    co      us     light   
328  10/11/2001 05:15            denver    co      us     light   
344  10/11/2003 20:00         frederick    co      us   unknown   

    duration (seconds) duration (hours/min)  \
31                1800              30mins.   
43                  60             1 minute   
167               2700          45 minutes+   
180                600           10 minutes   
183                 15           15 seconds   
198                300            5 minutes   
203                 60                 1min   
252                 20     15 to 20 seconds   
328                900           15 minutes   
344                 10           10 seconds   

                                              comments date posted   latitude  \
31   very low clouds all different colors&#44search...   1/28/1999  38.758333   
43   Three Stars Begin to Move Randomly&#44 Stop&#4...   7/16/2006  40.015000   
167  Huge lighted cluster in the eastern sky.  ((NU...  11/28/2007  39.739167   
180                   UFO SPOTTED&#44 FLEMING COLORADO  11/28/2007  40.680000   
183  Hovering triangular shape seen with flashing l...  11/28/2007  39.885556   
198  In the northwest sky&#44 there were about seve...  10/31/2008  38.254444   
203  metallic egg shaped craft above crested butte&...  12/12/2009  38.869722   
252   Bright zig zagging light in the sky above norad.  10/30/2012  38.833889   
328                         Odd Lights in Colorado Sky  10/12/2001  39.739167   
344  Large flash outside the house rattling the win...  10/15/2003  40.099167   

      longitude  
31  -104.742500  
43  -105.270000  
167 -104.984167  
180 -102.838889  
183 -104.986667  
198 -104.608611  
203 -106.987222  
252 -104.820833  
328 -104.984167  
344 -104.936667

## How many sightings met these criteria?

In [ ]:
len(subset_2)

1524

**1524**